assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -479,6 +763,31 @@ describe("DiversionPage", () => {
     expect(screen.getByRole("button", { name: "Edit Detour" })).toBeVisible()
   })
 
+  test("'Share Detour Details' screen returns to editing screen when edit detour button is clicked", async () => {
+    const { container } = render(<DiversionPage />)
+
+    fireEvent.click(originalRouteShape.get(container))
+
+    fireEvent.click(originalRouteShape.get(container))
+
+    await userEvent.click(finishDetourButton.get())
+
+    await waitFor(() => {
+      expect(screen.getByRole("button", { name: "Edit Detour" })).toBeVisible()
+    })
+
+    await userEvent.click(screen.getByRole("button", { name: "Edit Detour" }))
+
+    await waitFor(() => {
+      expect(
+        screen.queryByRole("button", { name: "Edit Detour" })
+      ).not.toBeInTheDocument()
+      expect(
+        screen.getByRole("button", { name: "Finish Detour" })
+      ).toBeVisible()
+    })
```

Comment:
suggestion: does this work?
```suggestion
    await userEvent.click(await screen.findByRole("button", { name: "Edit Detour" }))

```

Reply:
I think Github may have gotten confused about what you were actually suggesting - can you re-suggest based on the diff as it currently is?

Reply:
https://github.com/mbta/skate/pull/2580/files/d34c0a0dfa11cdb8e586f6110f5e685e47fb701f#r1603782294